Kernel/Syscalls/sleep.cpp

```diff
@@ -35,8 +36,12 @@ int Process::sys$usleep(useconds_t usec)
     if (!usec)
         return 0;
     u64 wakeup_time = Thread::current()->sleep(usec * TimeManagement::the().ticks_per_second() / 1000000);
-    if (wakeup_time > g_uptime)
-        return -EINTR;
+    if (wakeup_time > g_uptime) {
+        u64 ticks_left_until_original_wakeup_time = wakeup_time - g_uptime;
+        u64 dt = ticks_left_until_original_wakeup_time / (TimeManagement::the().ticks_per_second() * 1000000);
+        // TODO - can we make this syscall return 64 bits instead of 32 bits?
```

Comment:
There already is a 64 bit sleep-like syscall: `clock_nanosleep` :P

In theory, we could even remove `sys$sleep` and `sys$usleep` and implement the LibC functions just through `sys$clock_nanosleep`.

Reply:
Indeed, we don't need 3 separate syscalls. I've removed `sys$sleep` and `sys$usleep` since everything can be implemented in terms of `sys$clock_nanosleep` with `CLOCK_MONOTONIC`